eng/Publishing.props

```diff
@@ -0,0 +1,40 @@
+<Project>
+
+  <PropertyGroup>
+    <PublishDependsOnTargets>$(PublishDependsOnTargets);CollectPackageChecksumFiles</PublishDependsOnTargets>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageChecksumFile Include="$(ArtifactsShippingPackagesDir)**\*.sha512" IsShipping="true" />
+    <PackageChecksumFile Include="$(ArtifactsNonShippingPackagesDir)**\*.sha512" IsShipping="false" />
+  </ItemGroup>
+
+  <Target Name="CollectPackageChecksumFiles">
+    <!-- Invoke the _GetVersionInfo custom target in this project in order to get the build version information.
+         This is necessary due to the Arcade's Publish.proj not including Arcade SDK, which provides the
+         calcuated version information. -->
+    <MSBuild Projects="$(RepoRoot)src\version.proj"
+        Targets="_GetVersionInfo"
+        SkipNonexistentProjects="false">
+      <Output TaskParameter="TargetOutputs" ItemName="_VersionInfo" />
+    </MSBuild>
+
+    <!-- Extract the build version information from the above project. -->
+    <PropertyGroup>
+      <_Version>@(_VersionInfo->'%(Version)')</_Version>
```

Comment:
Only concern I have here is we already have differing versioning schemes: the client is not the same version as other tools (major is different). Monitor will also use a different version scheme all-together. This feels like it would belong to the project itself (although it would be really cumbersome to call it per package).

Reply:
The way that I understand it is that we want all of the assets that are uploaded to blob storage to be grouped by what build they came from, which is the "diagnostics/$(_Version)" portion of the relative path. Thus the version component should be the "version" of the collection of assets, more specifically the default calculated version. Then each asset needs to be versioned according to how it versions itself.

This way, you can easily find all of the blob assets that are part of the same build and still have each asset versioned independently.

Reply:
I think the way I have it now is okay, but might need to be revisited when I enable aka.ms linking for the latest version of the packages. Since the latest link will need to be per servicing line, the relative blob path will need to contain something that isn't a interpreted as a version (see https://github.com/dotnet/arcade/blob/1c72df206204ff5675035abd98dd3ba30ffd20aa/src/Microsoft.DotNet.Build.Tasks.Feed/src/PublishArtifactsInManifest.cs#L1210) so that it remains in the in shortened url e.g. "diagnostics/5.0.0-preview#/monitor5.0/dotnet-monitor-5.0.0-preview#.nupkg.version" would be shortened to "diagnostics/monitor5.0/dotnet-monitor.nupkg.version" for the "latest" link. We'd put in the "monitor5.0" portion to make sure that we have a latest link for the 5.0.X line.

Reply:
I've been thinking about this more and I think we shouldn't incorporate a full version value as the root because of what you've pointed out (there are disparate versions for different packages). What I'm thinking is we need to have each package be part of a "release group" which is denoted by some moniker and the major and minor version of the package.

For example, dotnet-monitor would be part of the "monitor5.0" group, client library would be part of the "client0.2" group, and the others would be part of the "release5.0" group (or whatever moniker we want for the "other" category). Each package project can declare the moniker portion of the group name and we can automatically calculate the version portion based on the major and minor version of the package. Projects that do not declare the moniker would be added to the release/other group.

Just taking dotnet-monitor as the example from here on, if its version is 5.0.#-preview#, then its:
- release group is "monitor5.0"
- sha file blob path is "diagnostics/monitor5.0/dotnet-monitor-5.0.#-preview#.nupkg.sha512"
- version file blob path is "diagnostics/monitor5.0/dotnet-monitor-5.0.#-preview#.nupkg.version"
- latest version link is "diagnostics/monitor5.0/dotnet-monitor.nupkg.version"

The dotnet-monitor.nupkg.version link would point to the dotnet-monitor-5.0.#-preview#.nupkg.version file, which has contents of "5.0.#-preview#". This allows other processes (e.g. dotnet-docker nightly updates) to easily look at this latest link for the version and lookup the content of the sha512 for the checksum without having to know the build version from which it was created. It also allows dotnet-monitor to have servicing lines based on the major and minor versions such that this file would point to the latest version within the 5.0.X servicing line, thus allowing for easy lookup of the latest version for each servicing line.